lib/ui/plugins.dart

```diff
@@ -0,0 +1,35 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of dart.ui;
+
+/// Functionality for Flutter plugin authors.
+abstract class PluginUtilities {
+  /// Get a handle to a named top-level or static callback function which can
+  /// be easily passed between isolates.
+  ///
+  /// `callback` must not be null.
+  ///
+  /// Returns an integer that can be provided to
+  /// [PluginUtilities.getCallbackFromHandle] to retrieve a tear-off of the
+  /// original callback. If `callback` is not a top-level or static function,
+  /// null is returned.
+  static int getCallbackHandle(Function callback) {
```

Comment:
rather than return an int, you could return an object with a single field that's an int. That way you get better type checking and the APIs end up self-documenting more.

Reply:
The main reason I hadn't done that in the first place was that the object would need to be serialized and then deserialized by the plugin author when being passed over the method channels. If that's not a problem, I'll go ahead and add described object.

Reply:
Well they're going to have to serialize and deserialize the int, it's barely any more effort for them to do that using the int property of an object they're holding instead of just an int they're holding.

Reply:
Fair enough. Added in the latest commit.